In [4]:
import pandas as pd


In [5]:
data = {
    'Company': ['Apple', 'Apple', 'Apple', 'Samsung', 'Samsung', 'Samsung'],
    'Year': [2021, 2022, 2023, 2021, 2022, 2023],
    'Total Revenue': [300000, 320000, 340000, 280000, 290000, 310000],
    'Net Income': [50000, 52000, 55000, 46000, 47000, 49000]
}

df = pd.DataFrame(data)
df


Company  Year  Total Revenue  Net Income
0    Apple  2021         300000       50000
1    Apple  2022         320000       52000
2    Apple  2023         340000       55000
3  Samsung  2021         280000       46000
4  Samsung  2022         290000       47000
5  Samsung  2023         310000       49000

In [6]:
# Sort data by company and year
df = df.sort_values(by=['Company', 'Year'])

# Calculate Revenue Growth (%)
df['Revenue Growth (%)'] = df.groupby('Company')['Total Revenue'].pct_change() * 100

# Calculate Net Income Growth (%)
df['Net Income Growth (%)'] = df.groupby('Company')['Net Income'].pct_change() * 100

df


Company  Year  Total Revenue  Net Income  Revenue Growth (%)  \
0    Apple  2021         300000       50000                 NaN   
1    Apple  2022         320000       52000            6.666667   
2    Apple  2023         340000       55000            6.250000   
3  Samsung  2021         280000       46000                 NaN   
4  Samsung  2022         290000       47000            3.571429   
5  Samsung  2023         310000       49000            6.896552   

   Net Income Growth (%)  
0                    NaN  
1               4.000000  
2               5.769231  
3                    NaN  
4               2.173913  
5               4.255319

# 📊 Financial Trend Analysis using pandas

This report analyzes the financial performance of two companies (Apple and Samsung) over 3 years using Python and pandas.

## Methodology
- Simulated financial data was created.
- The percentage change in revenue and net income was calculated using `pct_change()` grouped by company.
- The goal was to identify year-over-year growth trends.

## Key Observations
- Apple shows steady revenue growth: 6.67% in 2022 and 6.25% in 2023.
- Samsung also shows positive trends: 3.57% in 2022 and 6.90% in 2023.
- Both companies have consistent net income growth, indicating financial stability.

## Conclusion
Pandas makes it easy to perform grouped calculations and analyze company trends over time.
This basic analysis can be expanded with more companies and additional financial metrics in the future.
